src/typing/nullSafety.ml

```diff
@@ -1455,10 +1455,15 @@ class expr_checker mode immediate_execution report =
 			Check if specified expressions can be passed to a call which expects `types`.
 		*)
 		method private check_args callee args types =
+			let rec expr_with_unsafe_meta e =
+				match e.eexpr with
+					| TMeta ((Meta.NullSafety, [(EConst (Ident "Off"), _)], _), e) -> true
+					| TParenthesis e -> expr_with_unsafe_meta e
+					| _ -> false in
 			let rec traverse arg_num args types meta =
 				match (args, types, meta) with
 					| (arg :: args, (arg_name, optional, t) :: types, arg_meta :: meta) ->
-						let unsafe_argument = contains_unsafe_meta arg_meta in
+						let unsafe_argument = contains_unsafe_meta arg_meta || expr_with_unsafe_meta arg in
```

Comment:
Why do you need that?
Couldn't you just do `if (@:nullSafety(Off) f(v))` ? (which also happens to be at least somewhat readable)

Reply:
This will disable null safety for inlined function code, see:
```haxe
final f = [1, 2, null, 3].find(item -> {
	item++; // should error
	item == 2;
});
```
Current change only casts argument type instead.

Reply:
:confused:  this both feels a bit hacky on compiler side and very ugly on haxe side imo